frmts/ogcapi/gdalogcapidataset.cpp

```diff
@@ -1670,7 +1695,12 @@ bool OGCAPIDataset::InitWithTilesAPI(GDALOpenInfo *poOpenInfo,
     if (!DownloadJSon(osTilingSchemeURL.c_str(), oDoc, nullptr,
                       MEDIA_TYPE_JSON))
         return false;
-    auto tms = gdal::TileMatrixSet::parse(oDoc.SaveAsString().c_str());
+
+    // Attempts to find the uri for a well-known TMS; if it does not work, it will send the entire document
+    const auto uri = oDoc.GetRoot().GetString("uri");
+
+    auto tms = gdal::TileMatrixSet::parse(
```

Comment:
@jerstlouis for the time being, I am just checking if it can get an uri, and if not it defaults to what was there before (e.g.: sending the entire document). At least this is now working for well-known uris.

Reply:
@jerstlouis I got that, but I don't want to address this refactoring in this PR.